delfin/drivers/utils/ssh_client.py

```diff
@@ -0,0 +1,147 @@
+        else:
+            path = os.getcwd()
+            filename = path + '\\' + self.ssh_host + 'known_hosts'
+            file = open(filename, 'w')
+            file.write(self.ssh_private_key)
+            file.close()
+            self.ssh.load_host_keys(filename)
+
+        self.ssh.connect(hostname=self.ssh_host, port=self.ssh_port,
+                         username=self.ssh_username,
+                         password=self.ssh_password,
+                         timeout=self.ssh_conn_timeout)
+
+    def execCommand(self, command_str):
+        result = None
+        try:
+            if command_str is not None:
+                if self.ssh is not None:
+                    stdin, stdout, stderr = self.ssh.exec_command(command_str)
+                    res, err = stdout.read(), stderr.read()
+                    re = res if res else err
+                    result = re.decode()
+        except Exception as e:
+            LOG.error(e)
+            result = e
+        return result
+
+    def close(self):
+        try:
+            if self.ssh is not None:
+                # Close connection
+                self.ssh.close()
+                self.ssh = None
+        except Exception as e:
+            LOG.error(e)
+
+    def doexec(self, context, command_str):
+        """Clear alert from storage system."""
+        re = None
+        try:
+            if command_str is not None:
+                self.connect()
+                re = self.execCommand(command_str)
+        except paramiko.AuthenticationException as ae:
+            LOG.error('doexec Authentication error:{}'.format(ae))
+            raise exception.InvalidUsernameOrPassword()
+        except Exception as e:
+            LOG.error('doexec InvalidUsernameOrPassword error:{}'.format(e))
+            if 'WSAETIMEDOUT' in str(e):
+                raise exception.SSHConnectTimeout()
+            elif 'No authentication methods available' in str(e) \
+                    or 'Authentication failed' in str(e):
+                raise exception.SSHInvalidUsernameOrPassword()
+            elif 'not a valid RSA private key file' in str(e):
+                raise exception.InvalidPrivateKey()
+            elif 'not found in known_hosts' in str(e):
+                raise exception.SSHNotFoundKnownHosts(self.ssh_host)
+            else:
+                raise exception.SSHException()
+
+        finally:
+            self.close()
+        return re
+
+    def login(self, context):
+        """Test SSH connection """
+        version = ''
+        try:
+            command_str = 'showwsapi'
+            re = self.doexec(context, command_str)
+            wsapiinfos = re.split('\n')
+            version = self.get_version(context, wsapiinfos)
+        except Exception as e:
+            LOG.error('login error:{}'.format(e))
+            raise e
+        return version
+
+    def get_version(self, context, wsapiinfos):
```

Comment:
Only generic code that can be used across different drivers should be outside the specific driver folder